### PR TITLE
Wrap diagnostics notification in collection flag check.

### PR DESCRIPTION
**Pull request by ryanwilson**
Some of the diagnostics notifications were missed and not covered by
the data collection flag.

### CODE REVIEW

File: Example/Core/Tests/FIRAppTest.m [hunk 1]
```diff
@@ -715,6 +716,28 @@ - (void)testGlobalDataCollectionNoDiagnosticsSent {
...
 
+- (void)testGlobalDataCollectionNoDiagnosticsSentForNoOptions {
+  // Add an observer for the diagnostics notification - both with and without an object to ensure it
+  // catches it either way. Currently no object is sent, but in the future that could change.
+  [self.notificationCenter addMockObserver:self.observerMock
+                                      name:kFIRAppDiagnosticsNotification
+                                    object:nil];
+  [self.notificationCenter addMockObserver:self.observerMock
+                                      name:kFIRAppDiagnosticsNotification
+                                    object:OCMOCK_ANY];
```
Comment (paulb777) [inline Example/Core/Tests/FIRAppTest.m]: Does OCMOCK_ANY include nil?

File: Firebase/Core/FIRApp.m [hunk 2]
```diff
@@ -274,13 +280,15 @@ + (void)addAppToAppDictionary:(FIRApp *)app {
...
   if ([app configureCore]) {
     sAllApps[app.name] = app;
-    [[NSNotificationCenter defaultCenter]
-        postNotificationName:kFIRAppDiagnosticsNotification
-                      object:nil
-                    userInfo:@{
-                      kFIRAppDiagnosticsConfigurationTypeKey : @(FIRConfigTypeCore),
-                      kFIRAppDiagnosticsFIRAppKey : app
-                    }];
+    if ([app isDataCollectionDefaultEnabled]) {
```
Comment (paulb777) [inline Firebase/Core/FIRApp.m]: It seems like this should only be done in configureCore - but that could be a separate refactor?

File: Example/Core/Tests/FIRAppTest.m (hunk 1 above)
Comment (ryanwilson) [reply Example/Core/Tests/FIRAppTest.m]: Will test to see if that's the case!

File: Firebase/Core/FIRApp.m (hunk 2 above)
Comment (ryanwilson) [reply Firebase/Core/FIRApp.m]: I'll try to refactor it now.

File: Example/Core/Tests/FIRAppTest.m (hunk 1 above)
Comment (ryanwilson) [reply Example/Core/Tests/FIRAppTest.m]: Good call,  it does include nil. I'll make sure we don't duplicate this elsewhere in a different PR (grabbed this from a different location).

File: Firebase/Core/FIRApp.m (hunk 2 above)
Comment (ryanwilson) [reply Firebase/Core/FIRApp.m]: Done.

File: Example/Core/Tests/FIRAppTest.m (hunk 1 above)
Comment (ryanwilson) [reply Example/Core/Tests/FIRAppTest.m]: Done.

**Comment by ryanwilson** [comment]
Thanks, looking into it now.

**Pull request merged**